app/controllers/transactions/transaction-detail-redirect.controller.js

```diff
@@ -1,23 +1,23 @@
 'use strict'
 
-const paths = require('../../paths')
 const { userServicesContainsGatewayAccount } = require('../../utils/permissions')
 const Ledger = require('../../services/clients/ledger.client')
 const { renderErrorView } = require('../../utils/response.js')
 const router = require('../../routes')
+const formatAccountPathsFor = require('../../utils/format-account-paths-for')
 
 const defaultMsg = 'Error processing transaction view'
 const notFound = 'Charge not found'
 
-module.exports = async (req, res) => {
+module.exports = async function redirectToTransactionDetail (req, res) {
   const chargeId = req.params.chargeId
   try {
     let charge = await Ledger.transactionWithAccountOverride(chargeId)
     if (userServicesContainsGatewayAccount(charge.gateway_account_id, req.user)) {
       req.gateway_account.currentGatewayAccountId = charge.gateway_account_id
       req.session = { ...req.session, backLink: req.header('Referer') }
       charge = null
-      res.redirect(302, router.generateRoute(paths.transactions.detail, { chargeId }))
+      res.redirect(302, router.generateRoute(formatAccountPathsFor(router.paths.account.transactions.detail, req.account.external_id), { chargeId }))
```

Comment:
No need for `router.generateRoute` here, the format account paths works generically similar to the original formatter, it should be possible to:

```suggestion
      res.redirect(302, formatAccountPathsFor(router.paths.account.transactions.detail, req.account.external_id, chargeId))
```

A little bit less to understand and potentially then able to remove the dependency on router. I think this is true of all of the formatters in this PR.

Reply:
Good spot - looks cleaner.
Done.